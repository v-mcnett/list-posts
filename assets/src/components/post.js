import moment from 'moment';

/**
 * Post Component.
 *
 * @param {string} postTitle - Current post title.
 * @param {Integer} postId - Current post ID
 * @param {string|boolean} featured_image - Posts featured image
 * @param {string|boolean} excerpt - Post excerpt
 * @param {string|boolean} link - Link to full post
 * @param {string|boolean} author name - Author post
 * @returns {*} Post HTML.
 */

 export const Post = ({ title: { rendered: postTitle } = {}, id: postId, featured_image = false, excerpt: { rendered: postExcerpt } = {}, link = '', author_name = '', date_gmt }) => (
  
	<article className="post" id={postId}>
        <h3 className="post__title">{postTitle}</h3>
        <p className="post__byline">Published{' '} 
        { date_gmt &&
            <time dateTime={moment(date_gmt).utc().format()}>
                {moment(date_gmt).local().format('MMMM DD, Y')}{' '}
            </time>  
        }
         by {author_name} </p>
		<figure className="post__figure" style={{ backgroundImage: `url(${featured_image})` }}>
		</figure>
        <div className="post__excerpt" dangerouslySetInnerHTML={{__html: postExcerpt.trim()}} />

        <a href={link} className="post__morelink">Continue reading</a>
        
	</article>
);