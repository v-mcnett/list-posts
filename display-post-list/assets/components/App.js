import React, { Component } from "react";
import { PostList } from './PostList';
import { Pagination } from './Pagination';
import axios from "axios";

class App extends Component {
  constructor(props) {
    super( ...arguments );
      this.props = props;
      this.state = {
			  posts: [],
        loading: false,
        currentPage: 1, 
        totalPages: 0,
        selectedPostCount: parseInt(window.reactObj.selectedPostCount) 
      };
        
      this.onChangePage = this.onChangePage.bind(this);
   // this.removeAnimation = this.removeAnimation.bind(this);
  }
  componentDidMount() {
    this.setState({ loading: true });
    this.getPosts(this.state.selectedPostCount, this.state.currentPage);
  }

  onChangePage(value) {

    this.setState({ 
      loading: true,  
      currentPage: parseInt(value)
    });

   this.getPosts(this.state.selectedPostCount, value);
  
  }

  getPosts(per_page, current_page) {
    const wpHeaders = {
      headers: {
        'Access-Control-Allow-Origin': '*',
        'Access-Control-Expose-Headers': 'x-wp-total'
      }
    };

    // Call to word press API to get posts 
    axios(`/wp-json/wp/v2/posts?per_page=${per_page}&page=${current_page}`, wpHeaders)
    .then((posts) => {
      this.setState({ 
        loading: false,
        selectedPostCount: parseInt(per_page),
        currentPage: parseInt(current_page),
        totalPages: parseInt(posts.headers['x-wp-totalpages']),
        posts: posts.data,
        });
    } ).catch((error) => {
      console.log(error);
    });
  }
  
  render() {
    return (
      <section>
        <PostList posts={this.state.posts} />
        <Pagination onChangePage={this.onChangePage} currentPage={this.state.currentPage} totalPages={this.state.totalPages}/> 
      </section>
    );
  }
}

export default App;
