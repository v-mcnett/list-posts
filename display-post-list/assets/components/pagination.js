import React, { Component } from 'react';

/**
 * Pagination Component
 * 
 * @param object props - Component props.
 * @returns next and previous links, if applicable 
 */
export class Pagination extends Component {
    constructor(props) {
        super(props);
        this.onClickPrevious = this.onClickPrevious.bind(this);
        this.onClickNext = this.onClickNext.bind(this);
    }

    onClickPrevious(e) {
        const { onChangePage } = this.props;
        onChangePage(this.props.currentPage - 1 );
    }

    onClickNext(e) {
        const { onChangePage } = this.props;
        onChangePage(this.props.currentPage + 1 );
    }

    render () {
        const { totalPages, currentPage } = this.props;
        let nextLink, previousLink;

        // No pagination necessary if less than or equal to 1 pag 
        if (totalPages <= 1) {
            return null;
        }

        // Previous link only needed after page 1
        if (currentPage > 1) {
            previousLink = <div><a className="pagination__previous" onClick={this.onClickPrevious} >Previous</a></div>
            ;
        } else {
            previousLink = <div></div>;
        }

        // Next link not needed on last page 
        if (currentPage < totalPages) {
            nextLink =  <div><a className="pagination__next" onClick={this.onClickNext}>Next</a></div>;
        } else {
            nextLink = <div></div>;
        }

        return (
            <div className="pagination">
                { previousLink }
                { nextLink }
            </div>
        );
    }
}