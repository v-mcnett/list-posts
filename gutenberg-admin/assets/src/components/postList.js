import { Post } from './Post';

/**
 * PostList Component
 * 
 * @param object props - Component props.
 * @returns HTML of posts
 */
export const PostList = props => {
	const { loading = false, posts = []} = props;

	if (loading) {
		return <p className="post-list-loading">Loading posts...</p>;
	}

	if ( ! posts || posts.length < 1 ) {
		return <p className="post-list-none">No posts found, please try again.</p>
	}

	return (
		<div className="post-list">
			{posts.map((post) => <Post key={post.id} {...post}  />)}
		</div>
	);
};