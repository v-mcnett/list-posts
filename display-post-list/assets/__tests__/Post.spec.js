import React from "react";
import renderer from 'react-test-renderer';
import { Post } from "../components/post";

describe("Post component", () => {
  test("Matches the snapshot", () => {
    const post = renderer.create(<Post />);
    expect(post.toJSON()).toMatchSnapshot();
  });
});