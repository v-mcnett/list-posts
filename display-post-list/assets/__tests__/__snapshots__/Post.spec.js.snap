// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post component Matches the snapshot 1`] = `
<article
  className="post"
>
  <h3
    className="post__title"
  />
  <p
    className="post__byline"
  >
    Published
     
     by 
    
     
  </p>
  <figure
    className="post__figure"
    style={
      Object {
        "backgroundImage": "url(false)",
      }
    }
  />
  <div
    className="post__excerpt"
    dangerouslySetInnerHTML={
      Object {
        "__html": undefined,
      }
    }
  />
  <a
    className="post__morelink"
    href=""
  >
    Continue reading
  </a>
</article>
`;
